### ISSUE TITLE
[typescript-resolvers] Better support for enum, mappers and internal values

### ISSUE THREAD

**Issue by dotansimha** [dotansimha/graphql-code-generator#3985]
When dealing with custom enum values, for example with internal values: https://www.graphql-tools.com/docs/scalars/#internal-values , codegen need to generate signature for the enum resolvers, and also use the mapped values in `input` types. 
When trying to combine it with `enumValues` it makes more errors, because then it will replace the original enum with the mapped one, without having the runtime resolvers for that enum. 

Reproduction (and added example for missing parts): https://codesandbox.io/s/hungry-benz-3kd2m?file=/resolvers.ts 

The actual missing parts are:

- `typescript-resolvers` should generate this generic type when enums are in use in the schema:

```ts
// T is the type of the enum (typeof MyEnum)
// AllowedValues by default is any, but should be replaced with the mapper value, if set for the enum
type EnumResolverSignature<T, AllowedValues = any> = {
  [key in keyof T]?: AllowedValues
};
```

- `typescript-resolvers` should generate a resolvers signature for enums by name:

Without `mappers`:

```ts
type Resolvers {
 // ... more resolvers ...
 MyEnum: EnumResolverSignature<typeof MyEnum>;
}
```

With `mappers` set:

```ts
type Resolvers {
 // ... more resolvers ...
 MyEnum: EnumResolverSignature<typeof MyEnum, MyCustomEnum>;
}
```

- `typescript-resolvers` should use `Omit` to replace enum fields with mapped values, also for input types, when it's set. We need to find an easy way to manage that, because if there are no implementation for the enum internal values, this might no be needed to be done.

**Comment by dotansimha** [comment]
Just a small fix: in case of explicit values for `enumValues`, we do need to generate enum resolvers, but in a different structure that forces the specific value of each enum value. Fixed it 👍 Also, added support for doing the same with `mappers` set for the enum, not just `enumValues`.

Also, fixed in for `flow-resolvers` plugin.

The latest alpha is available as `1.15.3-alpha-30577e3d.0`.

**Comment by dotansimha** [comment]
Available in `v1.15.3`.
